model.py

```diff
@@ -11388,6 +11390,20 @@ def site_configuration_has_changed(_db, timeout=1):
     number of seconds since the last site configuration change was
     recorded.
     """
+    has_lock = site_configuration_has_changed_lock.acquire(blocking=False)
```

Comment:
Since blocking=False this will immediately return, True if we were able to get the thread lock and False if not.

Since the goal of this code is to make sure a timestamp is updated, it doesn't have to run if another thread is currently running it.